sys/fmt/fmt.c

```diff
@@ -137,6 +137,10 @@ uint32_t scn_u32_dec(const char *str, size_t n)
 
 void print(const char *s, size_t n)
 {
+#ifdef __WITH_AVRLIBC__
+    /* AVR's libc doesn't offer write(), so use fwrite() instead */
+    fwrite(s, n, 1, stdout);
```

Comment:
Well, write() is usually a system call, while fwrite() is a libc layer on top of that.

write() might fail for many reasons, like interruption by signal, or the system might decide to only write half of the data as it pleases.

fwrite() uses write(), but usually adds some buffering. Because of that, it has to deal with incomplete write() calls internally, as the user of fwrite cannot know how e.g., the caching is implemented.

For that reason I assume that fwrite usually succeeds unless the underlying file has a hard error (e.g., it gets closed, at disk full, or there's a hardware failure), which I also assume won't happen for the AVR's serial port.